### PR TITLE
Use Vec<u8> instead of String

### PR DESCRIPTION
**Pull request by majecty**

### CODE REVIEW

File: remote-trait-object/src/packet.rs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::fmt;
+
+pub struct SlotId(u32);
+
+impl fmt::Debug for SlotId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("SlotId").field("id", &self.0).field("type", &self.get_type()).finish()
```
Comment (junha1) [inline remote-trait-object/src/packet.rs:23]: https://doc.rust-lang.org/std/fmt/trait.Debug.html
"Generally speaking, you should just derive a Debug implementation."
I think this kind of implementation would belong to `Display`.

File: remote-trait-object/src/packet.rs [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    pub fn to_owned(&self) -> Packet {
+        Packet::new_from_buffer(self.buffer.to_vec())
+    }
+}
+
+pub struct Packet {
+    buffer: Vec<u8>,
+}
+
+impl fmt::Debug for Packet {
```
Comment (junha1) [inline remote-trait-object/src/packet.rs:164]: ditto

File: remote-trait-object/src/packet.rs [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+const SLOT_CALL_OR_RETURN_INDICATOR: SlotId = SlotId(1000);
+
+// FIXME: repr(C) is not a reliable encoding method.
+// We need to fix the endianness of binary data.
+#[repr(C)]
+struct PacketHeader {
+    pub slot: SlotId,
+    pub target_service_name: [u8; 100],
+    pub method: [u8; 100],
```
Comment (junha1) [inline remote-trait-object/src/packet.rs:69]: Please put `FIXME: use an integer index for the service and method`.

File: remote-trait-object/src/packet.rs (hunk 1 above)
Comment (majecty) [reply remote-trait-object/src/packet.rs]: Thanks. I chaged it.

File: remote-trait-object/src/packet.rs (hunk 2 above)
Comment (majecty) [reply remote-trait-object/src/packet.rs]: Thanks. I chaged it.

File: remote-trait-object/src/packet.rs (hunk 3 above)
Comment (majecty) [reply remote-trait-object/src/packet.rs:72]: I added a comment.

**Comment by majecty** [comment]
@junha1 I added a comment on the append_data function.

**Pull request merged**